### PR TITLE
Periodically clean old rows from time_series table

### LINKED ISSUE #6645: Delete old rows from time_series on a schedule
The `time_series` table currently grows without bounds. We should add a cron job that deletes rows older than e.g. 24 hours periodically.

### PR DESCRIPTION
**Pull request by nwalters512**
Closes #6645.

### CODE REVIEW

File: cron/cleanTimeSeries.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const { callbackify } = require('util');
```
Comment (nwalters512) [inline cron/cleanTimeSeries.js:1]:
```suggestion
// @ts-check
const { callbackify } = require('util');
```

File: cron/cleanTimeSeries.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+const logger = require('../lib/logger');
+const sqldb = require('../prairielib/lib/sql-db');
+const sqlLoader = require('../prairielib/lib/sql-loader');
+
+const sql = sqlLoader.loadSqlEquiv(__filename);
+
+module.exports.run = callbackify(async () => {
+  const result = await sqldb.queryAsync(sql.clean_time_series);
```
Comment (nwalters512) [inline cron/cleanTimeSeries.js:11]:
```suggestion
  const result = await sqldb.queryAsync(sql.clean_time_series, {});
```

File: cron/index.js [hunk 3]
```diff
@@ -104,6 +104,11 @@ module.exports = {
         intervalSec:
           config.cronOverrideAllIntervalsSec || config.cronIntervalWorkspaceHostTransitionsSec,
       },
+      {
+        name: 'cleanTimeSeries',
+        module: require('./cleanTimeSeries'),
+        intervalSec: 'daily',
```
Comment (mwest1066) [inline cron/index.js:110]: Can we add a config variable for this interval and made it default to something like 20 minutes? I don't want to lock the table for very long when we have other writers (the load reporting cron jobs, in particular).

**Review by mwest1066 — APPROVED** [review]

File: cron/cleanTimeSeries.sql [hunk 4]
```diff
@@ -0,0 +1,2 @@
+-- BLOCK clean_time_series
+DELETE FROM time_series WHERE date < now() - interval '24 hours';
```
[the remark this answers is not in the document]
Comment (nwalters512) [reply cron/cleanTimeSeries.sql]: Done in 5d24a37.

File: cron/index.js (hunk 3 above)
Comment (nwalters512) [reply cron/index.js]: Done in 5d24a37 (I opted to do it every 10 minutes).

File: cron/cleanTimeSeries.sql [hunk 5]
```diff
@@ -0,0 +1,10 @@
+-- BLOCK clean_time_series
+WITH selected_ids AS (
+    SELECT id
+    FROM time_series
+    WHERE date < now() - make_interval(secs => $retention_period_sec)
```
Comment (mwest1066) [inline cron/cleanTimeSeries.sql:5]: Add `ORDER BY date` (or `id`?) to improve deletion locality? I'm not sure this really matters?

**Review by mwest1066 — APPROVED** [review]

File: cron/cleanTimeSeries.sql (hunk 5 above)
Comment (nwalters512) [reply cron/cleanTimeSeries.sql:5]: I'm not sure either... let's wait until we get #6651 deployed so we can `EXPLAIN` and see how the query planner handles this?

File: lib/config.js [hunk 6]
```diff
@@ -76,7 +76,22 @@ config.cronIntervalWorkspaceTimeoutStopSec = 60;
...
+/**
+ * Controls how much history is retained when removing old rows
+ * from the `time_series` table in the database.
+ */
+config.timeSeriesRetentionPeriodSec = 24 * 60 * 60; // 1 day
+/**
+ * Controls how many rows are deleted in each batch when removing old rows
+ * from the `time_series` table in the database.
+ */
+config.cleanTimeSeriesBatchSize = 20_000;
```
[the remark this answers is not in the document]
Comment (nwalters512) [reply lib/config.js:94]: Yeah, I'd be happy to increase this!

File: cron/cleanTimeSeries.sql (hunk 5 above)
Comment (mwest1066) [reply cron/cleanTimeSeries.sql:5]: Sounds good to me!

Comment (mwest1066) [reply cron/cleanTimeSeries.sql:5]: Although with your new plan to just delete all of them I guess this no longer matters.

Comment (nwalters512) [reply cron/cleanTimeSeries.sql]: Agreed!

File: lib/config.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (nwalters512) [reply lib/config.js]: After talking with Matt, we decided to remove the limit entirely.

File: lib/config.js [hunk 7]
```diff
@@ -76,7 +76,17 @@ config.cronIntervalWorkspaceTimeoutStopSec = 60;
 config.cronIntervalWorkspaceTimeoutWarnSec = 60;
 config.cronIntervalWorkspaceHostLoadsSec = 10;
 config.cronIntervalWorkspaceHostTransitionsSec = 10;
+/**
+ * Note that this should be set such that we're always deleting more rows
+ * than we're inserting.
+ */
```
Comment (nwalters512) [inline lib/config.js:82]:
```suggestion
```

**Pull request merged**